File: src/lib/json.go
package lib

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"os"

	"github.com/nt92/memex/src/schema"
)

const docsJsonPath = "./../output/docs.json"
const indexJsonPath = "./../output/index.json"

func CreateDocsJson(records schema.RecordInfo) {
	json_data, _ := json.Marshal(records)

	file, _ := os.OpenFile(
		docsJsonPath,
		os.O_WRONLY|os.O_TRUNC|os.O_CREATE,
		0666,
	)

	defer file.Close()
	file.Write(json_data)
}

func CreateIndexJson(tokenIndex schema.TokenIndex) {
	json_data, _ := json.Marshal(tokenIndex)

	file, _ := os.OpenFile(
		indexJsonPath,
		os.O_WRONLY|os.O_TRUNC|os.O_CREATE,
		0666,
	)

	defer file.Close()
	file.Write(json_data)
}

func SearchIndex(searchTerm string) {
	indexFile, _ := os.Open(indexJsonPath)
	defer indexFile.Close()
	byteValue, _ := ioutil.ReadAll(indexFile)

	var tokenIndex schema.TokenIndex
	err := json.Unmarshal([]byte(byteValue), &tokenIndex)
	if err != nil {
		log.Fatal(err)
	}

	searchRecords(tokenIndex[searchTerm])
}

func searchRecords(IDs []string) {
	recordsFile, _ := os.Open(docsJsonPath)
	defer recordsFile.Close()
	byteValue, _ := ioutil.ReadAll(recordsFile)

	var recordInfo schema.RecordInfo
	err := json.Unmarshal([]byte(byteValue), &recordInfo)
	if err != nil {
		log.Fatal(err)
	}

	for _, ID := range IDs {
		recordContent := recordInfo[ID].Content
		recordTime := recordInfo[ID].Time
		fmt.Println(TimestampToDate(recordTime), recordContent)
	}
}

File: src/lib/json.py
import json

from src.schema.record_info import RecordInfo
from src.schema.token_index import TokenIndex

docs_json_path = './db/docs.json'
index_json_path = './db/index.json'


def create_docs_json(record: RecordInfo):
    with open(docs_json_path, 'w') as output:
        json.dump(record.to_json(), output)


def create_index_json(token_index: TokenIndex):
    with open(index_json_path, 'w') as output:
        json.dump(token_index.to_json(), output)


def search_index(query: str):
    with open(index_json_path, 'r') as index_file:
        index_json = json.load(index_file)
        index = TokenIndex(**index_json)
        print(index.index.values())
